bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateServiceTest.java

```diff
@@ -318,6 +319,39 @@ public void testCancelFirmwareUpdate() {
             verify(handler3, times(1)).cancel();
         });
     }
+    
+    @Test
+    public void testCancelFirmwareUpdateIntheMiddleOfUpdate() {
+        final long stepsTime = 200;
+        final int numberOfSteps = SEQUENCE.length;
+        final AtomicBoolean isUpdateFinished = new AtomicBoolean(false);
+        
+        doAnswer(invocation -> {
+            ProgressCallback progressCallback = (ProgressCallback) invocation.getArguments()[1];
+            progressCallback.defineSequence(SEQUENCE);
+            
+            // Simulate update steps with delay
+            for(int updateStepsCount=0; updateStepsCount < numberOfSteps; updateStepsCount++) {
```

Comment:
There are some spaces missing. Please use the formatter of the Eclipse SmartHome IDE.

Reply:
Thanks, I will apply the formatter :)